jscomp/ounit_tests/ounit_utf8_test.ml

```diff
@@ -19,5 +43,81 @@ let suites =
         __LOC__ >:: begin fun _ -> 
             Ext_utf8.decode_utf8_string
             "" =~ []
-        end
+        end;
+        __LOC__ >:: begin fun _ ->
+            Ext_string.append "Hell" 'o' =~ "Hello"
+        end;
+        __LOC__ >:: begin fun _ ->
+            Ast_utf8_string.consume_text "Hello $world" 0 =~ ("Hello ", 6)
+        end;
+        __LOC__ >:: begin fun _ -> 
+            let s, new_index = Ast_utf8_string.consume_text "Hello \\$world" 0 in
+            let _ = s =~ "Hello \\$world" in
```

Comment:
I am a bit confused here, what is the expected behavior? Note here `"Hello \\$world"` in fact equals to `{j|Hello \$world|j}` since the OCaml parser will do some processing and escape the `\\$` to '\$'. Thanks.